lib/http.js

```diff
@@ -31,12 +31,46 @@ exports.request = function(options, cb) {
   return new ClientRequest(options, cb);
 };
 
+exports.requestAsync = function(options) {
+  var resolveRequest = null;
+  const getResponse = new Promise((resolve) => {
+    resolveRequest = resolve;
+  });
+  const request = exports.request(options, (res) => {
+    resolveRequest(res);
+  });
+  return {
+    request,
+    response: getResponse
```

Comment:
I'm not sure it's quite consistent when `request` returns `{request, response}` instead of `request` object. For me `request.response.then(res => ...)` makes more sense. But yes, it will not work with the destructuring assignment, so it's debatable.